addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +113,144 @@ def _get_duration_from_tracking(self, trackings):
+
+    If the rotting_threshold_days field is not defined on the tracked module,
+    or if the value of rotting_threshold_days is 0,
+    then the resource will never rot.
+    """
+
+    rotting_days = fields.Integer('Days Rotting', help='Day count since this resource was last updated',
+        compute='_compute_rotting')
+    is_rotting = fields.Boolean('Rotting', compute='_compute_rotting', search='_search_is_rotting')
+
+    # To override in inheriting models
+    def _get_rotting_depends_fields(self):
+        if self._is_rotting_enabled():
+            return ['date_last_stage_update', f'{self._track_duration_field}.rotting_threshold_days']
+        return []
+
+    # To override in inheriting models
+    def _get_rotting_domain(self):
+        if self._is_rotting_enabled():
+            return [
+                (f'{self._track_duration_field}.rotting_threshold_days', '!=', 0),
+            ]
+        return [fields.Domain.FALSE]
+
+    def _is_rotting_enabled(self):
+        return hasattr(self, '_track_duration_field') and 'rotting_threshold_days' in self[self._track_duration_field] and (
+            not self  # api.model call
+            or any(stage.rotting_threshold_days for stage in self[self._track_duration_field])
+        ) and 'date_last_stage_update' in self
+
+    @api.depends(lambda self: self._get_rotting_depends_fields())
+    def _compute_rotting(self):
+        could_rot_records = self.filtered_domain(self._get_rotting_domain())
+        clean_records = (self - could_rot_records)
+        for stage, records in could_rot_records.grouped(self._track_duration_field).items():
+            rotting_records = records.filtered(lambda record: fields.Datetime.now() > record.date_last_stage_update + timedelta(days=stage.rotting_threshold_days))
+            rotting_records.is_rotting = True
+            for record in rotting_records:
+                record.rotting_days = (fields.Datetime.now() - record.date_last_stage_update).days
+            clean_records += (records - rotting_records)
+        clean_records.is_rotting = False
+        clean_records.rotting_days = 0
+
+    def _search_is_rotting(self, operator, value):
+        """
+        :param operator
+        :param value
+        :return domain
+
+        Override this search method to complete the search domain for is_rotting field
+        """
+        if operator not in ['in', 'not in']:
+            raise UserError(_('Operation not supported'))
+        if not self._is_rotting_enabled():
+            return [fields.Domain.FALSE]
+
+        # review note:
+        # hey wouldn't it be cool if query objects actually gave you access to their alias table names?
+        # I am currently building on data whose alias names I don't know for sure and have to infer based on how
+        # they're build at time of writing, but if the alias names are then determined by something that isn't make_alias()
+        # in the future then this implementation breaks
```

Comment:
Well I hope it won't happen, this is low level and people should not play with it. Would stick on using current code from query class, even if a bit low level and not that clear, have to admit